pkg/descheduler/node/node.go

```diff
@@ -218,7 +218,12 @@ func fitsRequest(nodeIndexer podutil.GetPodsAssignedToNodeFunc, pod *v1.Pod, nod
 		resourceNames = append(resourceNames, name)
 	}
 
-	availableResources, err := nodeAvailableResources(nodeIndexer, node, resourceNames)
+	availableResources, err := nodeAvailableResources(nodeIndexer, node, resourceNames,
+		func(pod *v1.Pod) (v1.ResourceList, error) {
+			req, _ := utils.PodRequestsAndLimits(pod)
```

Comment:
I suppose that the new future implementations of this function should ensure that there is no overlap between the pod requests and the other sources.

Reply:
It's every invoker's responsibility to make sure the returned resource utilization is valid. E.g. metrics might do aggregation of resources to provide a meaningful value.